src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1504,6 +1505,20 @@ class kernel_impl : public std::enable_shared_from_this<kernel_impl>
     return data;  // no skipping
   }
 
+  static xrt::module
+  get_module(const xrt::hw_context& ctx, const std::string& kname)
+  {
+    // ELF use case, identify module from ctx that has given kernel name
+    // kernel name will be of format - <kernel_name>:<index>
+    auto i = kname.find(":");
+    if (i == std::string::npos) {
+      // default case
+      return xrt_core::hw_context_int::get_module(ctx, kname.substr(0, kname.size()));
```

Comment:
Isn't this just `kname`?  As in 
```suggestion
      return xrt_core::hw_context_int::get_module(ctx, kname);
```